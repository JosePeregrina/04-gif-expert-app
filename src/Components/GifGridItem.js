import React from 'react';
import PropTypes from 'prop-types';

function GifGridItem({ title, url }) {
  return (
    <div className='card animate__animated animate__fadeIn animate__slow'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
}

export default GifGridItem;

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
