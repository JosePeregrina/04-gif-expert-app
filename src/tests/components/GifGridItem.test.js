import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../Components/GifGridItem';

describe('El componente GifGridItem debe...', () => {
  const title = 'HunterXHunter';
  const url = 'http://algo.com';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('cargar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect( url.includes('http://') ).toBe(true);
  });
  
  test('tener un parrafo con el <title>', () => {
      const p = wrapper.find('p');

      expect( p.text().trim() ).toBe(title);
  });
  
  test('tener la imagen con su <url> y su alt con el valor <title>', () => {
      const img = wrapper.find( 'img' );
      expect( img.prop( 'src' ) ).toBe( url );
      expect( img.prop('alt') ).toBe( title )
  })

  test('debe tener la clase css "animate__fadeIn"', () => {
      const div = wrapper.find('div');

      expect( div.hasClass('animate__fadeIn') ).toBe( true );
  })
  
  

});
