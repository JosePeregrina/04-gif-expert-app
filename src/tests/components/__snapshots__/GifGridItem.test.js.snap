// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente GifGridItem debe... cargar correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__slow"
>
  <img
    alt="HunterXHunter"
    src="http://algo.com"
  />
  <p>
    HunterXHunter
  </p>
</div>
`;
